crates/circuit/src/circuit_instruction.rs

```diff
@@ -129,7 +129,7 @@ impl CircuitInstruction {
         )
     }
 
-    fn __getstate__(&self, py: Python<'_>) -> PyObject {
+    pub fn __getstate__(&self, py: Python<'_>) -> PyObject {
```

Comment:
It probably doesn't matter, but this makes me nervous - it feels very weird to have the `__getstate__` methods be officially marked "public access".  They're not typically even meant to be called directly in Python space.  Do we ever need to call them from Rust space?

Also, this is definitely pre-existing, but I don't actually think that this class needs `__getstate__` or `__setstate__` methods at all - `__getnewargs__` is already doing all the stateful work.  It shouldn't have much of an effect, since `pickle` will only be pickling the args one time each anyway, but it's still just a little more inefficient than we need.

(If you want to remove them in a separate PR so we can see if there's any `copy` improvement in `QuantumCircuit`, I'm also cool with it.)

Reply:
Addressed in 84835e3ba3966e362e568749eabcbf172219d778.